sql/catalyst/src/main/scala/org/apache/spark/sql/Encoder.scala

```diff
@@ -37,7 +38,33 @@ trait Encoder[T] extends Serializable {
   def clsTag: ClassTag[T]
 }
 
+/**
+ * Methods for creating encoders.
+ */
 object Encoders {
+
+  /**
+   * (Scala-specific) Creates an encoder that serializes objects of type T using Kryo.
+   * This encoder maps T into a single byte array (binary) field.
+   */
+  def kryo[T: ClassTag]: Encoder[T] = {
```

Comment:
How about putting this method into `ExpressionEncoder`? I think `Encoders` is only used at java side as the lack of implicit magic.